src/Http/Http.Extensions/src/UriHelper.cs

```diff
@@ -204,24 +203,7 @@ public static string GetEncodedPathAndQuery(this HttpRequest request)
     /// suitable only for display.</returns>
     public static string GetDisplayUrl(this HttpRequest request)
     {
-        var scheme = request.Scheme ?? string.Empty;
-        var host = request.Host.Value ?? string.Empty;
-        var pathBase = request.PathBase.Value ?? string.Empty;
-        var path = request.Path.Value ?? string.Empty;
-        var queryString = request.QueryString.Value ?? string.Empty;
-
-        // PERF: Calculate string length to allocate correct buffer size for StringBuilder.
-        var length = scheme.Length + SchemeDelimiter.Length + host.Length
-            + pathBase.Length + path.Length + queryString.Length;
-
-        return new StringBuilder(length)
-            .Append(scheme)
-            .Append(SchemeDelimiter)
-            .Append(host)
-            .Append(pathBase)
-            .Append(path)
-            .Append(queryString)
-            .ToString();
+        return string.Concat((ReadOnlySpan<string?>)[request.Scheme, SchemeDelimiter, request.Host.Value, request.PathBase.Value, request.Path.Value, request.QueryString.Value]);
```

Comment:
Is this casting really necessary to pick the correct overload? I'd hoped `params ReadOnlySpan<string?>` would be picked over `params string?[]` so you could call it like this:

```csharp
string.Concat(request.Scheme, SchemeDelimiter, request.Host.Value, request.PathBase.Value, request.Path.Value, request.QueryString.Value)
```

Reply:
It was, last time I checked. Have you got different results?

Reply:
I hadn't, it just struck me as weird 😅 

Tried it in SharpLab now (it doesn't yet have the `ReadOnlySpan<string?>` overload though: https://sharplab.io/#v2:EYLgtghglgdgPgAQEwEYCwAoBBmABM3AYVwG9NcL88EUAGXAWQAob6BnAYwAsBTMHgDT46uLgHs2AFyGtcABwiSuAIQhtBw+gqUyRARwCuPAE4BPAJSlylGwgDsRMTA6KmnXvyEBVY1AB0AMrcfDwAIjwANlBgUJImQuJSQtpcyYoqahqGJhYA3NYUAL4FuCU4mo7OrgrGEGBsuABKPBAAJgDyMBGmAQowADwBkr4wAOYA/AB8uABuEBFGbJZkGDZr9pp+AKJgcpKmuZQlxatHpxTlsoROLpJMNXUNrOMA2gC6s/OLyyXrDqzbXb7Q4UY6YQpAA=

Reply:
Calling it with a collection expression (with or without the casting) seems to result in the exact same lowered code 😄

Reply:
It was introduced in preview6.

I think the plan for the compiler is to always choose the most performant option. But it's not there yet. Maybe in the next preview.

Reply:
```suggestion
        return string.Concat([request.Scheme, SchemeDelimiter, request.Host.Value, request.PathBase.Value, request.Path.Value, request.QueryString.Value]);
```

This should work now.

https://sharplab.io/#v2:EYLgtghglgdgPgAQIwAYB0BhA9jAxhAFwAoBtAIgDMssyAaAAjOAgCcyBdASgG4g